### PR TITLE
Edit profile

### PR DESCRIPTION
**Pull request by ShahzodK**
Create page Edit profile

### CODE REVIEW

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      </mat-form-field>
+      <mat-form-field appearance="fill">
+        <mat-label>{{'editProfile.login' | translate}}</mat-label>
+        <input type="text" matInput name="login" formControlName="login">
+        <!-- <mat-error *ngIf="hasEmailError">{{ getEmailErrorMessage() | translate }}</mat-error> -->
+      </mat-form-field>
+      <mat-form-field appearance="fill">
+        <mat-label>{{'editProfile.password' | translate}}</mat-label>
+        <input type="password" matInput name="password" formControlName="password">
+        <!-- <mat-error *ngIf="hasPasswordError">{{ getPasswordErrorMessage() | translate }}</mat-error> -->
```
Comment (AdiletMaksatuly) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.html:20]: Implement all error displaying methods that you commented out

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    this.api.deleteUser(this.userService.getUserId()).subscribe({
+      complete: () => {
+        this.store.dispatch(UserActions.resetUser());
+        this.router.navigateByUrl('home');
+      },
+    });
+  }
+
+  public submit(): void {
+    console.log('submitted');
```
Comment (AdiletMaksatuly) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts:76]: remove all console logs before committing

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    const id = this.userService.getUserId();
+    const name = this.editProfileForm.getRawValue().name;
+    const login = this.editProfileForm.getRawValue().login;
+    const password = this.editProfileForm.getRawValue().password;
+    if (this.editProfileForm.valid) {
```
Comment (AdiletMaksatuly) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts:81]: if (this.editProfileForm.invalid) return;

this.api.updateUser... etc

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+@Component({
+  selector: 'app-edit-profile-page',
+  templateUrl: './edit-profile-page.component.html',
+  styleUrls: ['./edit-profile-page.component.scss'],
+})
+export class EditProfilePageComponent implements OnInit {
+
+  constructor(
+    private store: Store,
+    private api: UserApiService,
```
Comment (AdiletMaksatuly) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts:21]: private userApi instead of api

**Review by AdiletMaksatuly — CHANGES_REQUESTED** [review]

File: src/app/shared/services/user.service.ts [hunk 5]
```diff
@@ -23,15 +23,16 @@ export class UserService {
 
   public check(): void {
     const id = this.getUserId();
-
     if (id.length === 0) {
       this.store.dispatch(resetUser());
     }
 
     this.http.get(`users/${id}`).subscribe({
```
Comment (nastyavolkm) [inline src/app/shared/services/user.service.ts:30]: it's a good practice, to write a returned type of response like for example
this.http.get<IUser>

than the is no need to use res as bla-bla

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    });
+    this.store.select(selectUserLogin).pipe(take(1)).subscribe({
+      next: (login) => this.loginValue = login,
+    });
+
+    this.editProfileForm.get('name')!.setValue(this.nameValue);
+    this.editProfileForm.get('login')!.setValue(this.loginValue);
+  }
+
+  public nameValue: string = '';
```
Comment (nastyavolkm) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts:38]: no need to write a type for primitive with initial value

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts [hunk 7]
```diff
@@ -58,13 +59,57 @@ export class EditProfilePageComponent implements OnInit {
       nonNullable: true,
       validators: [
         Validators.required,
-        passwordStrengthValidator(),
+        this.UserPasswordStrengthValidator(),
       ],
     }),
   });
 
+  public UserPasswordStrengthValidator(): ValidatorFn {
```
Comment (AdiletMaksatuly) [inline src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts:67]: what's the reason for creating new validator?

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.html (hunk 1 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.html]: done

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts (hunk 2 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts]: done

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts (hunk 3 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts]: done

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts (hunk 4 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts]: done

File: src/app/shared/services/user.service.ts (hunk 5 above)
Comment (ShahzodK) [reply src/app/shared/services/user.service.ts]: done

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts (hunk 6 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts]: done

File: src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts (hunk 7 above)
Comment (ShahzodK) [reply src/app/user-profile/pages/edit-profile-page/edit-profile-page.component.ts]: deleted and implemented already created validator

**Review by AdiletMaksatuly — APPROVED** [review]

**Pull request merged**